woocommerce/woocommerce-gutenberg-products-block
# Product Gallery: Add a Product Image fallback

## What

Fixes #11940

## Why

This PR introduces a fallback for scenarios where the Product Image is not set, but some Product Gallery images are present. In such cases, a placeholder image will be displayed as the first image in the Product Gallery.

## Testing Instructions



_Please consider any edge cases this change may have, and also other areas of the product this may impact._

1. With a block theme, open `Appearance > Editor`.
2. Open `Templates > Single Product`.
3. Add the `Product Gallery` block.
5. Open a product page on the frontend with multiple product images set in the product gallery, but **no main Product Image set**. Confirm that the Product Gallery works as expected (Thumbnails, Next Previous Buttons and Pager) and that the first image is correctly set to the placeholder image.



## Screenshots or screencast



| Before | After |
| ------ | ----- |
|<img width="1043" alt="Moto_Guzzi_v7_–_newproductgallery-4" src="https://github.com/woocommerce/woocommerce-blocks/assets/905781/88af52a4-2e9d-43f8-a9a8-3b1c2df536cf">|<img width="1046" alt="Moto_Guzzi_v7_–_newproductgallery-3" src="https://github.com/woocommerce/woocommerce-blocks/assets/905781/d110630e-fd7b-4c01-ae23-6c90507df033">|

## WooCommerce Visibility


Required:


## Checklist

Required:


Conditional:


## Changelog


> Add suggested changelog entry here.
